src/Features/Core/Portable/NavigateTo/RoslynNavigateToItem.cs

```diff
@@ -189,25 +207,71 @@ private bool IsNamedType()
 
             ImmutableArray<TextSpan> INavigateToSearchResult.NameMatchSpans => _item.NameMatchSpans;
 
-            string INavigateToSearchResult.SecondarySort
+            string INavigateToSearchResult.SecondarySort => _secondarySort.Value;
+
+            private string ComputeSecondarySort()
             {
-                get
+                using var _ = ArrayBuilder<string>.GetInstance(out var parts);
+
+                // Ensure if all else is equal, that high-pri items (e.g. from the user's current file) come first
+                // before low pri items.  This only applies if things like the MatchKind are the same.  So we'll
+                // still show an exact match from another file before a substring match from the current file.
+                parts.Add(ComputeFolderDistance().ToString("X4"));
```

Comment:
this item (and the helper it calls) is the meat of hte change.

Reply:
it seems like if we gave a file path with the result this is something the platform could do entirely on their end.  I think they would have to do this work for any search provider implemented via LSP as well (unless we add support for more complex sorting there too).

Reply:
Agreed. Though a virtue of this is that we can do this in a non allocating fashion. Def harder (but possible) on the platform side.